code/controllers/subsystem/processing/quirks.dm

```diff
@@ -43,18 +43,21 @@ PROCESSING_SUBSYSTEM_DEF(quirks)
 
 /datum/controller/subsystem/processing/quirks/proc/AssignQuirks(datum/mind/user, client/cli, spawn_effects)
 	var/list/bad_quirks = list()
+	var/positive_quirks = 0
 	for(var/V in cli.prefs.all_quirks)
 		var/datum/quirk/Q = quirks[V]
 		if(Q)
 			user.add_quirk(Q, spawn_effects)
+			if(Q.quirk_value > 0) //Tally up the total of all of the positive quirks, future-proofing for when we re-add 2-point positive quirks
+				positive_quirks += Q.quirk_value
 		else
 			stack_trace("Invalid quirk \"[V]\" in client [cli.ckey] preferences. the game has reset their quirks automatically.")
 			bad_quirks += V
-	if(length(bad_quirks)) // negative & zero value = calculation good / positive quirk value = something's wrong
+	if(length(bad_quirks) || positive_quirks > MAX_POSITIVE_QUIRKS)
 		cli.prefs.all_quirks = list()
 		// save the new cleared quirks.
 		cli.prefs.mark_undatumized_dirty_character()
-		client_alert(cli, "You have one or more outdated quirks[length(bad_quirks) ? ": [english_list(bad_quirks)]" : ""]. Your eligible quirks are kept at this round, but your character preference has been reset. Please review them at any time.", "Oh, no!")
+		client_alert(cli, "You have invalid quirks: ([length(bad_quirks) ? ": [english_list(bad_quirks)]" : "Too many positive quirks"]). Your quirks are kept at this round, but your character preference has been reset. Please review them at any time.", "Oh, no!")
```

Comment:
```suggestion
		client_alert(cli, "You have invalid quirks: ([length(bad_quirks) ? english_list(bad_quirks) : "Too many positive quirks"]). Your quirks are kept at this round, but your character preference has been reset. Please review them at any time.", "Oh, no!")
```

Reply:
It's not clear which you want removed or why since you have removed the brackets but show parenthesis in your comment. Removing brackets will actively break the debugging message, and removing parenthesis will take away the wrapper for the message?

Without further changes my code should display one of these two messages:

`You have invalid quirks: (List of bad quirks). Your quirks are kept at this round...`
`You have invalid quirks: (Too many positive quirks). Your quirks are kept at this round...`

If I commit the changes you have sent to me the first message will read this:
`You have invalid quirks: (english list(bad_quirks)). Your quirks are kept at this round...`

Reply:
the way you're doing is `You have invalid quirks: : (badqhirk1, badquirk2)`